api/v1alpha1/author_webhook.go

```diff
@@ -0,0 +1,167 @@
+
+// RegisterAuthorWebhook registers the author webhook with the passed manager and log.
+func RegisterAuthorWebhook(mgr ctrl.Manager, log *logr.Logger) {
+	mgr.GetWebhookServer().Register("/mutate-appstudio-redhat-com-v1alpha1-author", &webhook.Admission{
+		Handler: &authorWebhook{
+			client: mgr.GetClient(),
+			log:    log,
+		},
+	})
+}
+
+// Handle creates an admission response for Release and ReleasePlan requests.
+func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	switch req.Kind.Kind {
+	case "Release":
+		return a.handleRelease(req)
+	case "ReleasePlan":
+		return a.handleReleasePlan(req)
+	default:
+		return admission.Errored(http.StatusInternalServerError,
+			fmt.Errorf("webhook tried to handle an unsupported resource: %s", req.Kind.Kind))
+	}
+}
+
+// handleRelease takes an incoming admission request and returns an admission response. Create requests
+// add an author label with the current user. Update requests are rejected if the author label is being
+// modified. All other requests are accepted without action.
+func (a *authorWebhook) handleRelease(req admission.Request) admission.Response {
+	release := &Release{}
+	err := json.Unmarshal(req.Object.Raw, release)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+	}
+
+	switch req.AdmissionRequest.Operation {
+	case admissionv1.Create:
+		a.setAuthorLabel(req.UserInfo.Username, release)
+
+		return a.patchResponse(req.Object.Raw, release)
+	case admissionv1.Update:
+		oldRelease := &Release{}
+		err := json.Unmarshal(req.OldObject.Raw, oldRelease)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+		}
+
+		if release.GetLabels()[metadata.AuthorLabel] != oldRelease.GetLabels()[metadata.AuthorLabel] {
+			return admission.Errored(http.StatusBadRequest, errors.New("release author label cannnot be updated"))
+		}
+	}
+	return admission.Allowed("Success")
+}
+
+// handleReleasePlan takes an incoming admission request and returns an admission response. If the
+// attribution label is set to true, the current user is set as the author. If the attribution label
+// is false, the author label is removed. The only exception is if the attribution label remains true
+// during an update and the author value is not modified, the previous author label remains.
+func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Response {
+	releasePlan := &ReleasePlan{}
+	err := json.Unmarshal(req.Object.Raw, releasePlan)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+	}
+	// Author label should not exist in any case if attribution is not true
+	if releasePlan.GetLabels()[metadata.AttributionLabel] != "true" {
+		delete(releasePlan.GetLabels(), metadata.AuthorLabel)
```

Comment:
I think that having a function with a proper name wrapping this line is more readable and maintainable. But you already know that as well.